### ISSUE TITLE
false positives: file->is_encrypted / file->is_link

### ISSUE THREAD

**Issue by fasterthanlime** [DrMcCoy/dmc_unrar#3]
In https://paraperipipiadoux.itch.io/seedust, the file at `SEEDUST v3.0/Files/levelbank/testmap/header.dat` is wrongly detected as encrypted *and* a hard/symbolic link, so dmc_unrar refuses to extract it.

However, if you simply comment out this switch:

https://github.com/DrMcCoy/dmc_unrar/blob/c414853a3c6cfc3b4bab963a794773e5116e6f6e/dmc_unrar.c#L2654-L2663

then everything extracts normally, and directory output matches unrar-nonfree 5.50.

Which leads me to think that this part of the code is fishy. I'll keep exploring!

---

btw, unrar v output:

```
    ..A....         8         8 100%  2018-08-09 14:13  A988DFF7  SEEDUST v3.0/Files/levelbank/testmap/header.dat
```

**Comment by DrMcCoy** [comment]
Yeah, there's a bug in that extra_data parsing loop. Tracing...

**Comment by DrMcCoy** [comment]
Should be fixed with 094381c47aea6c2b4e5c77fd11f8a4c7c68e7096 :)
